bip-XXXX.mediawiki

```diff
@@ -0,0 +1,137 @@
+<pre>
+  BIP: ?
+  Layer: Consensus (soft fork)
+  Title: Disallow 64 byte transactions
+  Author: Chris Stewart <stewart.chris1234@gmail.com>
+  Status: Draft
+  Type: Specification
+  License: BSD-3-Clause
+  Created: ?
+</pre>
+
+==Abstract==
+
+This BIP describes the rationale for disallowing transactions that are serialized to 64 bytes without the transaction's witness. 
+We describe the weaknesses to the merkle tree included in bitcoin block headers, various exploits for those weaknesses.
+
+==Motivation==
+
+Bitcoin block headers include a commitment to the set of transactions in a given
+block, which is implemented by constructing a Merkle tree of transaction id’s
+(double-SHA256 hash of a transaction) and including the root of the tree in the
+block header. This in turn allows for proving to a Bitcoin light client that a
+given transaction is in a given block by providing a path through the tree to the
+transaction. However, Bitcoin’s particular construction of the Merkle tree has
+several security weaknesses, including at least two forms of block malleability
+that have an impact on the consensus logic of Bitcoin Core, and an attack on
+light clients, where an invalid transaction could be ”proven” to appear in a block
+by doing substantially less work than a SHA256 hash collision would require.
+This has been enforced by relay policy since 2018<ref>[https://github.com/bitcoin/bitcoin/pull/11423/commits/7485488e907e236133a016ba7064c89bf9ab6da3 PR #11423 disallows 64 byte transactions in bitcoin core relay]</ref>
+
+
+==Specification==
+
+This BIP disallows bitcoin transactions that are serialized to 64 bytes in length without it's witness.
+==Rationale==
+
+=== Block malleability ===
+
+64 byte transactions introduce block malleability. Malicious peers can construct consensus valid and invalid 64 byte
+transactions that have the same serialization as the concatenation of 2 leaves in the merkle tree.
```

Comment:
The 64-byte transaction can match to the concatenation of two neighbouring branches, not just two neighbouring leaves.

I think the "valid and invalid" bit is a little misleading -- if you want the hashes to match a valid 64B tx, then they have to give you a valid utxo as the prevout which requires the first hash to match 28 bytes (in counter and 27 bytes of the utxo's txid) which is cryptographically hard. That's also true for matching a coinbase tx (where the "txid" is all 0's).

Ending up with a plausible but invalid tx is the main attack here, in that you either convince a real node that a valid block is invalid, or you convince a light client that an invalid tx really exists and was confirmed.